backend/app/Services/LocationFetcher.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Services;
+
+use Illuminate\Http\Client\PendingRequest;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+
+class LocationFetcher
+{
+    protected const API_URL = "https://api.mapbox.com/geocoding/v5/mapbox.places/";
+
+    protected string $mapboxToken;
+    protected PendingRequest $request;
+    protected Collection $receivedData;
+
+    public function __construct()
+    {
+        $this->mapboxToken = env("VITE_MAPBOX_TOKEN");
```

Comment:
This should be config variable. Maybe create own config file like `internships.php` and start storing in them our config variables.

Reply:
I think I'll store these in `services` config, because it seems to contain stuff like external api tokens.